tests/unit/mesh/test_mesh.cpp

```diff
@@ -180,3 +180,61 @@ TEST_CASE("MakeSimplicial", "[Mesh]")
    // on the original mesh, but it doesn't happen for these test cases.
    REQUIRE(simplex_mesh.GetNE() == orig_mesh.GetNE()*factor);
 }
+
+TEST_CASE("MakeNurbs", "[Mesh]")
+{
+   Array<real_t> intervals_array({1, 1, 1, });
+   Vector intervals(intervals_array.GetData(), intervals_array.Size());
+   Array<int> continuity({-1, 1, 1, -1});
+   {
+      const KnotVector kv(2, intervals, continuity);
+      REQUIRE(kv.GetNE() == 3);
+      REQUIRE(kv.GetNCP() == 5);
+      REQUIRE(kv.GetOrder() == 2);
+      REQUIRE(kv.Size() == 8);
+   }
+   {
+      const KnotVector kv(3, intervals, continuity);
+      REQUIRE(kv.GetNE() == 3);
+      REQUIRE(kv.GetNCP() == 8);
+      REQUIRE(kv.GetOrder() == 3);
+      REQUIRE(kv.Size() == 12);
+   }
+   const KnotVector kv(2, intervals, continuity);
+   Array<NURBSPatch *> patches;
+   Array<real_t> grev_pts({0.0, 1.0/6.0, 0.5, 5.0/6.0, 1.0});
+
+   Array<real_t> pts(4 * kv.GetNCP() * kv.GetNCP() * kv.GetNCP());
+   int count = 0;
+   for (int k = 0; k < kv.GetNCP(); ++k)
+      for (int j = 0; j < kv.GetNCP(); ++j)
+         for (int i = 0; i < kv.GetNCP(); ++i)
+         {
+            pts[count + 0] = grev_pts[i];
+            pts[count + 1] = grev_pts[j];
+            pts[count + 2] = grev_pts[k];
+            pts[count + 3] = 1;
+            count += 4;
+         }
+   patches.Append(new NURBSPatch(&kv, &kv, &kv, 4, pts.GetData()));
+   Mesh patch_topology =
```

Comment:
Would it be useful to have a 2D test as well?

Reply:
I modified the test case to pull a lot of the checks out into a lambda expression, which is then applied to both 2D and 3D cases.  (I don't see a lot of lambdas in MFEM, but Google confirms it's part of C++11, so I believe it's considered allowable.)